website/content/en/docs/advanced-topics/heterogeneous.md

```diff
@@ -0,0 +1,400 @@
+If once you've determined which subset of architecture nodes you want to distribute your operator for, there are tools available to help you distribute your operator.
+
+OperatorSDK provides a validator to help check bundle configuration.
+```bash
+operator-sdk bundle validate ./bundle --select-optional name=multiarch
+```
+
+If you're planning to distribute the operator via OLM, you can find more information in the [OLM guide for supporting multiarch](https://olm.operatorframework.io/docs/advanced-tasks/ship-operator-supporting-multiarch/).
+
+### Setting the Supported Architectures Operator & Operand Workloads
+[Kubernetes provides a mechanism](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/) called `nodeAffinity` which can be used to limit the possible node targets where a pod can be scheduled. The following example can be used to update a `PodSpec` or `PodTemplateSpec` to prevent the scheduling of pods on nodes of incompatible architectures. Here we compare the `kubernetes.io/arch` key set on the node to ensure that the value matches one of the supported architectures. This assumes that the referenced image points to a manifest list of images for each of the architectures supported. It is important to remember that `nodeAffinity` must be set anywhere a container reference is defined, including `Pod`, `Deployment`, `DaemonSet`, `StatefulSet`, or any other object that defines a `PodSpec` or `PodTemplateSpec`.
+
+```yaml
+affinity:
+  nodeAffinity:
+    requiredDuringSchedulingIgnoredDuringExecution:
+      nodeSelectorTerms:
+      - matchExpressions:
+        - key: kubernetes.io/arch
+          operator: In
+          values:
+          - amd64
+          - arm64
+          - ppc64le
+          - s390x
+```
+
+The list of architecture values should only include the architectures the operator supports. The full list of possible `GOARCH` values is [available here](https://github.com/golang/go/blob/master/src/go/build/syslist.go). It should be noted that Kubernetes allows a user to specify both `requiredDuringSchedulingIgnoredDuringExecution` and `preferredDuringSchedulingIgnoredDuringExecution`. The syntax for each block is the same. At a minimum, an operator author should set the required terms since they protect against a pod being scheduled on an architecturally incompatible node. If the operator in question has much better performance on a subset of the supported architectures, it may also be prudent to set `preferredDuringSchedulingIgnoredDuringExecution` so that the optimal arches are selected first if they are available.
+
+#### Integrating with Operator Lifecycle Manager
+
+The Operator Lifecycle Manager (OLM) is often used to distribute operators via operator catalogs. The operator deployment object in an OLM integrated operator is defined in the `ClusterServiceVersion` yaml. It is important to remember to set the node affinity block in each of the `spec.template.spec` under `spec.install.deployments`.
```

Comment:
The CSV's `spec.install.deployments` block is automatically generated from the operator deployment generated by `kustomize build ./config`. So the correct place to add the node affinity stuff is in the `./config/manager/manager.yaml` file.

I wonder if we could separate the upstream, non-OLM concepts (e.g. manifest lists, setting node affinity, etc) from the OLM-specific bits (setting the CSV supported os/arch annotations)

Reply:
Follow up to previous comment: For those _not_ using OLM, the doc should have a clear set of things to do to ensure the operator/operands are setup with the correct manifest list images and node affinity configs. And that set of stuff has nothing to do with bundles or CSVs.

Then in this `Integrating with OLM` section, we could talk about what _extra_ stuff someone should do to build/validate a heterogeneous-compatible bundle.

Reply:
By the way, I just want to call to your attention that there's a whole separate docs section for integrating with OLM here: https://sdk.operatorframework.io/docs/olm-integration/

I'm not sure if the docs site already has most/all of the OLM-related stuff under that section or whether we've let OLM references spread throughout all the sections.

Either way, it might be good to take note of how other docs bring in the OLM integrations and do something similar for this one.